digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/util/AssetDataImporter.scala

```diff
@@ -301,6 +328,92 @@ class AssetDataImporter {
+          RoadTypeChangePoints(roadNumber, roadPartNumber, addressMChangeRoadType, RoadType.apply(roadTypeBefore), RoadType.apply(roadTypeAfter), ely)
+      }
+    }
+
+    val roadsWithSingleRoadType = conversionDatabase.withDynSession {
+      sql""" select distinct tie, aosa, tietyyppi, ely from VVH_TIEOSOITE_NYKY
+             where (tie, aosa) in(
+             SELECT gr.TIE, gr.AOSA FROM VVH_TIEOSOITE_NYKY gr  group by gr.tie, gr.aosa  having count(distinct gr.TIETYYPPI) = 1)
+              union
+            select distinct tie, aosa, tietyyppi, ely from vvh_tieosoite_taydentava
+            where (tie, aosa) in(
+            SELECT t.TIE, t.AOSA FROM vvh_tieosoite_taydentava t  group by t.tie, t.aosa  having count(distinct t.TIETYYPPI) = 1)
+            order by tie, aosa
+        """.as[(Long, Long, Long, Long)].list
+    }
+
+    withDynTransaction{
+      roadsWithSingleRoadType.foreach(road => {
+        updateRoadWithSingleRoadType(road._1, road._2, road._3, road._4)
+      })
+
+      roadsWithMultipleValues.foreach(road =>{
+        val addresses = RoadAddressDAO.fetchByRoadPart(road.roadNumber, road.roadPartNumber, true)
+        addresses.nonEmpty match {
+          case true => {
+            println(s"updating tie = ${road.roadNumber}, aosa = ${road.roadPartNumber}")
+            addresses.foreach(address =>{
+              roadType(roadsWithMultipleValues.filter { r=> r.roadNumber == road.roadNumber && r.roadPartNumber == road.roadPartNumber }, address) match {
+                case Left(roadType) =>{
+                  sqlu"""UPDATE ROAD_ADDRESS SET ROAD_TYPE = ${roadType.value}, ELY= ${road.elyCode} where ID = ${address.id} """.execute
+                }
+                case Right((addrM, roadTypeBefore, roadTypeAfter)) => {
+                  val roadLinkFromVVH = linkService.getCurrentAndComplementaryRoadLinksFromVVH(Set(address.linkId), false)
+                  val splittedRoadAddresses = splitRoadAddresses(address.copy(geometry = roadLinkFromVVH.head.geometry), addrM, roadTypeBefore, roadTypeAfter, road.elyCode)
+                  RoadAddressDAO.expireById(Set(address.id))
+                  RoadAddressDAO.create(splittedRoadAddresses)
+                }
+              }
+            })
+          }
+          case _ =>
+        }
+      })
+    }
+
+  }
+
+  def splitRoadAddresses(roadAddress : RoadAddress, addrMToSplit: Long, roadTypeBefore : RoadType, roadTypeAfter : RoadType, elyCode : Long) : Seq[RoadAddress] = {
+    val splitMValue = roadAddress.startMValue + (roadAddress.endMValue - roadAddress.startMValue) / (roadAddress.endAddrMValue - roadAddress.startAddrMValue) * (addrMToSplit - roadAddress.startAddrMValue)
+    println(s"Splitting road address id = ${roadAddress.id}, tie = ${roadAddress.roadNumber} and aosa = ${roadAddress.roadPartNumber}, on MValue = ${addrMToSplit}")
+    val roadAddressA = roadAddress.copy(
```

Comment:
Does not take into account the digitization direction: would put AgainstDigitization road addresses into wrong order

Reply:
In here, digitization affects MValues? I thought that is only related with geometry.

Reply:
It tells about the direction of growth.
Towards Digitization: startM = 0.0 <=> startMAddress, endM <=> endMAddress
Against Digitization: startM = 0.0 <=> endMAddress, endM <=> startMAddress